web/src/routes/(user)/search/+page.svelte

```diff
@@ -112,27 +112,35 @@
   };
 
   export const loadNextPage = async () => {
-    if (curPage == null || !term || (searchResultAssets && searchResultAssets.length >= MAX_ASSET_COUNT)) {
+    if (currentPage == null || !term || (searchResultAssets && searchResultAssets.length >= MAX_ASSET_COUNT)) {
       return;
     }
 
     await authenticate();
     let results: SearchResponseDto | null = null;
-    $page.url.searchParams.set('page', curPage.toString());
-    const res = await search({ ...$page.url.searchParams });
+    $page.url.searchParams.set('page', currentPage.toString());
+    const payload = $searchPayload;
+    let reponses: SearchResponseDto;
+
+    if (payload && 'query' in payload) {
+      reponses = await searchSmart({ smartSearchDto: { ...payload, page: parseInt(currentPage) } });
```

Comment:
The page here should be whatever is returned from the server. It shouldn't assume this is an int.

Reply:
The `page` property expect a number

Reply:
That's something that should change on the server side then to make it a string. I suppose we can change that later.